### DISCUSSION TITLE
CDKサンプルにSnapShotテストを付けてほしい

### DISCUSSION THREAD

**Discussion by koichimurakami** [mbc-net/mbc-cqrs-serverless#43] [Ideas]
CLIで作成するプロジェクトについてくるCDKサンプルにスナップショットテストを付けてほしいです。

システムに機能追加されるとインフラを変更する必要が出て来ます。
インフラの変更が意図したものかどうかを判断するためにはCDKにて作成するCloudFormationテンプレートがどのように変更されるか確認をしたいです。

そのため、CDKサンプルにスナップショットテストを作成し、CloudFormationテンプレートに変更があるかどうかを確認出来る様にして下さい。

**Comment by Tuan-Anh-Nguyen-pionero** [comment]
@koichimurakami cc@baquan1708 

Could I confirm this task with you?
Will the CDK Snapshot Test be executed in the pipeline instead of via GitHub Actions? 
If so, may I proceed with the test using your AWS account?
It mean i need to test to make sure function work well!
There may be costs incurred for testing services like EC2, NAT, RDS... estimated ~$20

**Comment by koichimurakami** [comment]
@Tuan-Anh-Nguyen-pionero 
I think execution in pipeline is better than GitHub Actions. Because Users will make infrastructure with cdk.
I can make environment for you. I'll contact directry.

**Reply by baquan1708** [reply]
@koichimurakami 
I've added a snapshot test and modified CDK infra to run test in the CodePipeline, as shown in the image below.

Please let me know when the test environment is ready.

In the meantime, I will investigate "CDK infrastructure diagram" in this [discussion](https://github.com/mbc-net/mbc-cqrs-serverless/discussions/44).

![codepipe-flow](https://github.com/user-attachments/assets/63f6a1b9-6ce2-49ce-b447-d76d49344e9c)

**Reply by baquan1708** [reply]
@koichimurakami cc: @Tuan-Anh-Nguyen-pionero 

I've just create a PR for this feature: https://github.com/mbc-net/mbc-cqrs-serverless/pull/46. When you have a moment, could you please take a look?
